Common/Orders/OrderJsonConverter.cs

```diff
@@ -194,5 +194,41 @@ private static Order CreateOrder(OrderType orderType, JObject jObject)
             }
             return order;
         }
+
+        /// <summary>
+        /// Creates a Time In Force of the correct type
+        /// </summary>
+        private static TimeInForce CreateTimeInForce(JToken timeInForce, JObject jObject)
+        {
+            // for backward-compatibility support deserialization of old JSON format
+            if (timeInForce is JValue)
+            {
+                var value = timeInForce.Value<int>();
+                if (value == (int)TimeInForceType.GoodTilDate)
+                {
+                    var expiry = jObject["DurationValue"].Value<DateTime>();
+                    return new GoodTilDateTimeInForce(expiry);
+                }
+
+                return (TimeInForceType)value;
+            }
+
+            var timeInForceType = (TimeInForceType)timeInForce["Type"].Value<int>();
+            switch (timeInForceType)
```

Comment:
maybe this switch statement can be removed/simplified by ensuring that we're serializing the `$type` to JSON. This will reduce coupling within the system and prevent potential errors from changes/additions to the TIF implementations.

Reply:
I'll try to handle serialization/deserialization with `$type`, although we'll still need a switch on a few numeric values (0,1,2) for backwards-compatibility.